GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -110,18 +118,50 @@ public bool TryGetLooseObjectId(string directoryName, string filePath, out strin
         /// Creates a pack file from loose objects
         /// </summary>
         /// <returns>The number of loose objects added to the pack file</returns>
-        public int CreateLooseObjectsPackFile()
+        public bool TryCreateLooseObjectsPackFile(out int objectsAddedToPack)
         {
-            int objectsAddedToPack = 0;
+            int localObjectCount = 0;
 
             GitProcess.Result result = this.RunGitCommand(
                 (process) => process.PackObjects(
                     "from-loose",
                     this.Context.Enlistment.GitObjectsRoot,
-                    (StreamWriter writer) => objectsAddedToPack = this.WriteLooseObjectIds(writer)),
+                    (StreamWriter writer) => localObjectCount = this.WriteLooseObjectIds(writer)),
                 nameof(GitProcess.PackObjects));
 
-            return objectsAddedToPack;
+            if (result.ExitCodeIsSuccess)
+            {
+                objectsAddedToPack = localObjectCount;
+                return true;
+            }
+            else
+            {
+                objectsAddedToPack = 0;
+                return false;
+            }
+        }
+
+        public int ClearCorruptLooseObjects()
+        {
+            int numBadObjects = 0;
+            foreach (string objectId in this.LooseObjectsBatch(this.MaxLooseObjectsInPack))
+            {
+                if (!this.Context.Repository.TryGetIsBlob(objectId, out bool isBlob) ||
+                    !isBlob)
+                {
+                    string objectFile = Path.Combine(
+                                                this.Context.Enlistment.GitObjectsRoot,
+                                                objectId.Substring(0, 2),
+                                                objectId.Substring(2, GVFSConstants.ShaStringLength - 2));

```

Comment:
Is it worth making the objectFile into a utility function?  

I saw this in another area of code:
            string blobPath = Path.Combine(
                this.enlistment.GitObjectsRoot,
                blobSha.Substring(0, 2),
                blobSha.Substring(2));

Reply:
I'm not ignoring this, just saving it for a later push.

Reply:
The previous use in `GitObjects` does too much to split out nicely (it creates the directory if needed and returns the folder AND file name). Hopefully it is enough that I made a method in `GitLooseObjectsStep` and created a unit test.